src/berry.c

```diff
@@ -0,0 +1,1254 @@
+#include "global.h"
+#include "berry.h"
+#include "main.h"
+#include "item.h"
+#include "items.h"
+#include "text.h"
+#include "rng.h"
+#include "event_data.h"
+#include "fieldmap.h"
+
+extern u8 FieldObjectGetBerryTreeId(u8 mapObjectId);
+extern void sub_8092EF0(u8 mapId, u8 mapNumber, u8 mapGroup);
+extern void CB2_ChooseBerry(void);
+extern const u8* GetFieldObjectScriptPointerForComparison(void);
+extern bool8 sub_8092E9C(u8, u8, u8);
+
+extern u16 gScriptItemId;
+
+extern const u8 BerryTreeScript[];
+extern const struct BerryTree gBlankBerryTree;
+
+#define BERRY_NAME_LENGTH 6
+
+#define FIRST_BERRY ITEM_CHERI_BERRY
+#define LAST_BERRY ITEM_ENIGMA_BERRY
+
+#ifdef ENGLISH
```

Comment:
You can get rid of this, since we're only building English Emerald.

Reply:
Can the defines stay though? Don't want to remove all the berry names DEFINES manually :P.